### PR TITLE
Add error message on wrong mapping

### LINKED ISSUE #1352: [6-dev] creating a collection with missing top "properties" object fails silently...
when attempting to create a collection with a invalid mapping, there is no, error, the collection is created but with an empty mapping:

``` JS 
kuzzle.collection.create('myIndex', 'myCollection', { 'a_field' : { 'type': 'keyword'}})
.catch((e) => console.log(e))
```
this leads to no error, collection not created que mapping empty

## Expected Behavior
I would expect to catch an error as the mapping I provided isn't correct

Also, admin console is misleading as it never shows this mandatory top level 'properties' object in mappings !!

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

:warning: Depends on https://github.com/kuzzleio/kuzzle/pull/1355

When creating a collection with a mapping or updating a mapping, the user can provide wrong properties (eg: `dinamic`, `foobar`) and Kuzzle will respond `200 OK` since ES just ignore theses properties.

This PR add a check on mapping properties (`dynamic`, `_meta` and `properties` are allowed)

Fix #1352 

### How should this be manually tested?

  - Step 1 : Use this script: 
```
const {
  Kuzzle,
  WebSocket
} = require('kuzzle-sdk');

const kuzzle = new Kuzzle(new WebSocket('localhost'));


(async () => {
  await kuzzle.connect();

  try {
    if (await kuzzle.index.exists('index')) {
      await kuzzle.index.delete('index');
    }
    await kuzzle.index.create('index');
    const rep = await kuzzle.collection.create('index', 'collection', { properties: {
      name: { type: 'keyword' },
      car: {
        meta: 'ee',
        properties: {}
      }
    } })
    console.log(rep)
  } catch (error) {
    console.error(error.message)
  } finally {
    kuzzle.disconnect()
  }
})();
```
  - Step 2 : You should see this error message: `Incorrect mapping mapping property "mapping.properties.car.meta". Did you mean "_meta" ?`

## Other changes

  - dont wrap string at 80 column

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [elasticsearch.js#](https://github.com/kuzzleio/kuzzle/blob/e99bfc54caccddf8eb57544a9e9c3bc86793e94f/lib/services/elasticsearch.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)
1. ![INFO][INFO] [pluginsManager.js#L811](https://github.com/kuzzleio/kuzzle/blob/e99bfc54caccddf8eb57544a9e9c3bc86793e94f/lib/api/core/plugins/pluginsManager.js#L811): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L811](https://github.com/kuzzleio/kuzzle/blob/65098693db7fe9c73fcad39725394f91c538f18b/lib/api/core/plugins/pluginsManager.js#L811): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/services/elasticsearch.js [hunk 1]
```diff
@@ -1382,6 +1393,27 @@ class ElasticSearch extends Service {
...
+
+  _checkMapping (mapping, path = [], check = true) {
+    const properties = Object.keys(mapping);
+
+    for (const property of properties) {
+      if (check && mappingProperties.indexOf(property) === -1) {
+        const currentPath = [...path, property].join('.');
+
+        throw new BadRequestError(
+          `Incorrect mapping mapping property "mapping.${currentPath}".` + didYouMean(property, mappingProperties));
```
Comment (scottinet) [inline lib/services/elasticsearch.js]:
```suggestion
          `Incorrect mapping property "mapping.${currentPath}".` + didYouMean(property, mappingProperties));
```

**Comment by scottinet** [comment]
Also you should probably sync this PR with @Yoann-Abbes 's https://github.com/kuzzleio/kuzzle/pull/1355

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L811](https://github.com/kuzzleio/kuzzle/blob/d1f28139e089a4d642f6c0bdbd8fabd2e1b4134c/lib/api/core/plugins/pluginsManager.js#L811): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by Aschen** [comment]
> Also you should probably sync this PR with @Yoann-Abbes's #1355

I merged https://github.com/kuzzleio/kuzzle/pull/1355 in my PR

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L811](https://github.com/kuzzleio/kuzzle/blob/b8e6d3a97887e194de610a7755365f32167bb7c2/lib/api/core/plugins/pluginsManager.js#L811): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [index.js#L101](https://github.com/kuzzleio/kuzzle/blob/b8e6d3a97887e194de610a7755365f32167bb7c2/lib/services/internalEngine/index.js#L101): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/plugins/pluginsManager.js [hunk 2]
```diff
@@ -634,8 +634,7 @@ class PluginsManager {
         if (typeof target !== 'function'
           && typeof plugin.object[target] !== 'function')
         {
-          let message = `Unable to configure pipe for event "${event}" with`
-          + ` provided method. "${target}" should be a plugin method name, or a function.`;
+          let message = `Unable to configure pipe for event "${event}" with provided method. "${target}" should be a plugin method name, or a function.`;
```
Comment (alexandrebouthinon) [inline lib/api/core/plugins/pluginsManager.js]: You maybe forget to apply the new wrapping rules :slightly_smiling_face:

Comment (Aschen) [reply lib/api/core/plugins/pluginsManager.js]: We didn't wrap the string

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L811](https://github.com/kuzzleio/kuzzle/blob/7b5c34715f977c7da0b7d03f742f337c1071651b/lib/api/core/plugins/pluginsManager.js#L811): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/plugins/pluginsManager.js [hunk 3]
```diff
@@ -662,8 +661,7 @@ class PluginsManager {
         if (typeof target !== 'function'
```
Comment (Yoann-Abbes) [inline lib/api/core/plugins/pluginsManager.js]: should not be
```
if (
typeof target !== 'function'
&& typeof plugin.object[target] !== 'function'
) {
?

File: lib/api/core/plugins/pluginsManager.js [hunk 4]
```diff
@@ -712,8 +708,7 @@ class PluginsManager {
         if (typeof description[action] !== 'function'
```
Comment (Yoann-Abbes) [inline lib/api/core/plugins/pluginsManager.js]: same of above

File: lib/services/elasticsearch.js [hunk 5]
```diff
@@ -206,7 +209,7 @@ class ElasticSearch extends Service {
 
     return this.client.get(esRequest)
       .then(result => {
-        if (result._source && 
+        if (result._source &&
```
Comment (Yoann-Abbes) [inline lib/services/elasticsearch.js]: same of above

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L813](https://github.com/kuzzleio/kuzzle/blob/b1290e733985e503c871d6920be7cc5e46e6084b/lib/api/core/plugins/pluginsManager.js#L813): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**